Update accounts-base.d.ts to reflect async await in EmailFields templ…

ToastHawaii opened the pull request:
Update accounts-base.d.ts to reflect async await in EmailFields templates.

<!--
First, 🌠 thank you 🌠 for taking the time to consider a contribution to Meteor!

Here are some important details to follow:

* ⏰ Your time is important
          To save your precious time, if the contribution you are making will take more
          than an hour, please make sure it has been discussed in an issue first.
          This is especially true for feature requests!
* 💡 Features
          Feature requests can be created and discussed by visiting:
          https://github.com/meteor/meteor/discussions
* 🕷 Bug fixes
          These can be created and discussed in this repository. When fixing a bug,
          please _try_ to add a test which verifies the fix.  If you cannot, you should
          still submit the PR but we may still ask you (and help you!) to create a test.
* 📖 Contribution guidelines
          Always follow https://github.com/meteor/meteor/blob/master/CONTRIBUTING.md
          when submitting a pull request.  Make sure existing tests still pass, and add
          tests for all new behavior.
* ✏️ Explain your pull request
          Describe the big picture of your changes here to communicate to what your
          pull request is meant to accomplish.  Provide 🔗 links 🔗 to associated issues!

We hope you will find this to be a positive experience!  Open source contribution can be intimidating and we hope to alleviate that pain as much as possible.  Without following these guidelines, you may be missing context that can help you succeed with your contribution, which is why we encourage discussion first.  Ultimately, there is no guarantee that we will be able to merge your pull-request, but by following these guidelines we can try to avoid disappointment.
-->

graemian committed:
Add instructions for enabling back gesture on iOS

ebroder committed:
Re-enable TypeScript self-test

The latest release of zodern:types is able to pull type declarations
from a Meteor checkout, if it's being used, instead of only from the
package repository (which may not contain the latest changes). This
means that we should be able to re-enable the TypeScript test, which
verifies that the TypeScript skeleton does, in fact, produce a project
which passes type checking.

(Re-enabling the test also picked up a few latent bugs in the type
declarations, so fix those as well)

zodern committed:
Add missing dependency on jquery

zodern committed:
Fix error generating xunit output

Grubba27 committed:
Merge branch 'devel' into fix/test-in-console

Grubba27 committed:
to version 1.2.7

StorytellerCZ committed:
Merge branch 'devel' into fix/test-in-console

nachocodoner committed:
Merge branch 'devel' into fix/test-in-console

mjcctech committed:
fix: displaying google maps in mobile application

dupontbertrand committed:
Update docs for upsert in ServiceConfiguration

Commit:
Update 3.forms-and-events.md

In the instructions to add onSubmit and onChange to TaskForm.jsx there was an unused import for TaskCollections. Removing this had no effect on the useability of the code in this step.

fredmaiaarantes committed:
Fix CLI and ecmascript links and add leonardoventurini:scss as the recommended scss package.

ToastHawaii committed:
Update accounts-base.d.ts to reflect async await in EmailFields templates.

italojs commented:
From what I found, those functions doenst retun a promise

https://github.com/meteor/meteor/blob/cad7de6cd39db0a47361637b280275dd367e7cb2/packages/accounts-password/email_templates.js#L29

https://github.com/meteor/meteor/blob/cad7de6cd39db0a47361637b280275dd367e7cb2/packages/accounts-password/email_templates.js#L34

https://github.com/meteor/meteor/blob/cad7de6cd39db0a47361637b280275dd367e7cb2/packages/accounts-password/email_templates.js#L39

maybe the correct would be remove the await from tthe calls
https://github.com/meteor/meteor/blob/7b775016bf457d9130247adcb5676bc43c59773d/packages/accounts-base/accounts_server.js#L1440

let me know if you found a place or usecase where it's returning a promise

ToastHawaii commented:
@italojs We overwrite the HTML template and other functions as described in this document https://docs.meteor.com/api/accounts.html#Accounts-emailTemplates and load some logos and other resources asynchronously with `Assets.getBinaryAsync`. 
After switching to Meteor v3, I don't know of any way to do this synchronously. So removing await at this point would be very bad for us as we would no longer be able to customise the email to our customers needs. 

So we need `await` at this point.

italojs commented on packages/accounts-base/accounts-base.d.ts:
```diff
@@ -10,10 +10,10 @@ export interface URLS {
-  subject?: ((user: Meteor.User) => string) | undefined;
-  text?: ((user: Meteor.User, url: string) => string) | undefined;
-  html?: ((user: Meteor.User, url: string) => string) | undefined;
+  from?: ((user: Meteor.User) => Promise<string>) | undefined;
```
could you return a Promise<string> | string in that function definition, please?

ToastHawaii committed:
Support Promise and no promise in EmailFields

ToastHawaii committed:
Merge branch 'devel' into patch-1

italojs reviewed (dismissed):
we will merge after the next release

carlosarraes committed:
feat: add support for dots in minimongo key names

Minimongo now allows dots (.) in MongoDB key names, aligning with modern
MongoDB support.

Fixes #13552

nachocodoner committed:
update caching strategy in meteor-selftest-windows workflow

nachocodoner committed:
use Windows tar instead of Git tar in workflow

italojs committed:
Merge branch 'devel' into fix-link-and-recommended-scss

italojs committed:
Merge branch 'devel' into patch-1

italojs committed:
Merge pull request #13428 from JacobKGilbert/patch-1

Update 3.forms-and-events.md

italojs committed:
Merge branch 'devel' into patch-2

italojs committed:
Merge pull request #13404 from dupontbertrand/patch-2

Update docs for upsert in ServiceConfiguration

italojs committed:
Merge branch 'devel' into fix-link-and-recommended-scss

italojs committed:
Merge pull request #13477 from meteor/fix-link-and-recommended-scss

docs: Fix links and add leonardoventurini:scss as recommended

nachocodoner committed:
Merge branch 'devel' into cache-windows-ci-check

nachocodoner committed:
re-run checks

nachocodoner committed:
re-run checks

nachocodoner committed:
clean

nachocodoner committed:
disable submodule update failure check in install.ps1

nachocodoner committed:
temp disable clean to verify cache speed

nachocodoner committed:
simplify cache paths in Windows self-test workflow

nachocodoner committed:
simplify cache paths in Windows self-test workflow

Grubba27 committed:
FEATURE: Remove `import React from 'react';` from skels

With our SWC changes, we can both remove this import line and add an example of swc configuration

nachocodoner committed:
skip temporary --get-ready

Grubba27 committed:
Merge pull request #14041 from meteor/feature/remove-react-import

FEATURE: Remove `import React from 'react';` from skels

nachocodoner committed:
refactor CI install script to adapt for CI environment

nachocodoner committed:
update workflow name to 'Windows Selftest'

nachocodoner committed:
refine Windows self-test workflow configuration

nachocodoner committed:
rename workflow file and update path references

nachocodoner committed:
fix path for windows-selftest.yml in workflow triggers

nachocodoner committed:
update cache key in Windows CI workflow

nachocodoner committed:
refactor dependency installation steps in Windows CI workflow

nachocodoner committed:
Merge pull request #14032 from meteor/cache-windows-ci-check

Windows CI check optimizations

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
remove wildcard branch trigger from Windows CI workflow

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
re-run checks

italojs committed:
Merge branch 'release-3.4.1' into fix/test-in-console

italojs committed:
Merge pull request #13000 from meteor/fix/test-in-console

Fix bugs with test-in-console

nachocodoner committed:
simplify cache key by removing package-lock.json hash

italojs committed:
Merge branch 'devel' into patch-1

italojs committed:
Merge pull request #12909 from graemian/patch-1

Add instructions for enabling back gesture on iOS

italojs committed:
merge branch

nachocodoner committed:
Merge branch 'devel' into release-3.4

italojs committed:
Merge branch 'release-3.4' into release-3.4.1

italojs committed:
test: make typescript template test async and await s.init()

italojs committed:
Merge branch 'release-3.4.1' into patch-1

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/meteor/meteor?pullRequest=13678) <br/>All committers have signed the CLA.
